django/db/models/query.py

```diff
@@ -937,10 +937,10 @@ def _batched_insert(self, objs, fields, batch_size):
                                              using=self.db)
 
     def _clone(self, klass=None, setup=False, **kwargs):
+        base_queryset_class = getattr(self, '_base_queryset_class', self.__class__)
         if klass is None:
             klass = self.__class__
-        elif not issubclass(self.__class__, klass):
-            base_queryset_class = getattr(self, '_base_queryset_class', self.__class__)
+        elif not (issubclass(base_queryset_class, klass) or issubclass(klass, base_queryset_class)):
```

Comment:
The goal state is `issubclass(klass, base_queryset_class)`.  The additional check is to handle the edge case of ValueListQuerySet being *upcasted* to ValuesQuerySet.